src/rules/if_change_then_change.rs

```diff
@@ -11,54 +12,103 @@ use regex::Regex;
 use crate::diagnostic;
 use crate::git;
 
+#[derive(Debug)]
+pub enum ThenChange {
+    RemoteFile(String),
+    RepoFile(PathBuf),
+    MissingIf,
+    MissingThen,
+    None,
+}
+
 #[derive(Debug)]
 pub struct IctcBlock {
     pub path: PathBuf,
     pub begin: i64,
     pub end: i64,
-    pub thenchange: PathBuf,
+    pub thenchange: ThenChange,
 }
 
-lazy_static::lazy_static! {
-    static ref RE_BEGIN: Regex = Regex::new(r"(?i)^\s*(//|#)\s*ifchange\s*$").unwrap();
-    static ref RE_END: Regex = Regex::new(r"(?i)^\s*(//|#)\s*thenchange\s*(.*)$").unwrap();
+impl IctcBlock {
+    fn get_range(&self) -> diagnostic::Range {
+        diagnostic::Range {
+            path: self.path.to_str().unwrap().to_string(),
+            start: diagnostic::Position {
+                line: self.begin as u64,
+                character: 0,
+            },
+            end: diagnostic::Position {
+                line: self.end as u64,
```

Comment:
`begin` and `end` should be defined as `Option<u64>` we're casting them from `u64` to `i64` and back to `u64` just so we can use `-1` to mark the field as `Option::None`